packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -544,8 +545,13 @@ function patchElementPropsAndAttrs(
     // value is set before type=radio.
     patchClassAttribute(oldVnode, vnode, renderer);
     patchStyleAttribute(oldVnode, vnode, renderer);
-    patchAttributes(oldVnode, vnode, renderer);
-    patchProps(oldVnode, vnode, renderer);
+
+    if (vnode.data.external) {
+        patchAttrUnlessProp(oldVnode, vnode, renderer);
+    } else {
+        patchAttributes(oldVnode, vnode, renderer);
+        patchProps(oldVnode, vnode, renderer);
```

Comment:
I think we should always do this... regardless of external or not. it is the compiler responsibility to not use any prop, only spreads, because that's intentionally forcing the system to use props. In which case, the patchProps will take care of the spread part of the equation only.